ykpack/src/types.rs

```diff
@@ -491,23 +547,18 @@ impl Statement {
 
         match self {
             Statement::Nop => (),
-            Statement::Assign(place, _rval) => place.push_maybe_defined_locals(&mut ret),
-            Statement::Enter(_target, args, dest, start_idx) => {
-                if let Some(dest) = dest {
-                    dest.push_maybe_defined_locals(&mut ret);
-                }
-                for idx in 0..args.len() {
-                    // + 1 to skip return value.
-                    ret.push(Local(start_idx + u32::try_from(idx).unwrap() + 1));
-                }
-            }
-            Statement::Leave => (),
+            Statement::IStore(dest, _src) => Self::maybe_push_local(&mut ret, dest.local()),
```

Comment:
Isn't `maybe_push_local` just `ret.map(|x| x.push(dest.local())` (or similar) in disguise?

Reply:
Yes. And perhaps it isn't even shorter. I'll kill it.

Sometimes helper functions turn out to be obfuscation functions.

Reply:
Hrm.

It doesn't work so well because  map is lazy and also returns a value:
```
555 | |             Statement::IStore(dest, _src) => dest.local().map(|l| ret.push(l)),
    | |                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found enum `std::option::Option`
```

Is there a way to ensure the map is consumed and to (concisely) ignore the return value?

Reply:
I've merged the cases here at least.

Reply:
OK.